src/ws-inline-edit/ws-inline-edit.js

```diff
@@ -0,0 +1,95 @@
+import {React, Component} from '../imports';
+require('./ws-inline-edit.scss');
+
+//  props: {
+//    text: 'Something'
+//  }
+
+/**
+ * This class describes a Preact component which renders a inline-edit element.
+ * The inline-edit component can be used everywhere in block where you want to have ability to change value.
+ * As an example you can use it in div blocks, rows, tables.
+ */
+export class WSInlineEdit extends Component {
+  /**
+   * @param {Object} props Peact props
+   * @constructor
+   */
+  constructor(props) {
+    super(props);
+    /**
+     * @type {Object}
+     */
+    this.state = {
+      isEditing: false,
+      text: props.text
+    };
+  }
+  /**
+   * Function that show input when you click on div and focus it
+   * @returns {void}
+   */
+  editElement() {
+    if (!this.state.isEditing) {
+      this.setState({isEditing: true}, () => {
+        this.editEl.focus();
+      });
+    }
+  }
+  /**
+   * Function that save text when click 'Enter' or cancel when click 'Escape' button
+   * @param {Object} e - click event
+   * @returns {Object}
+   */
+  keyAction(e) {
+    if (e.keyCode === 13) {
+      // Enter to save
+      this.setState({
+        text: e.target.value,
+        isEditing: false
+      });
+    } else if (e.keyCode === 27) {
+      // ESC to cancel
+      this.setState({isEditing: false});
+    }
+  }
+  /**
+   * Function that save text when input on blur and send text value to updating function
+   * @param {Object} e - click event
+   * @returns {Object}
+   */
+  blurAction(e) {
+    this.setState({
+      text: e.target.value,
+      isEditing: false
+    });
+    this.updating(e.target.value);
+  }
+  /**
+   * Function that return value for outside use
+   * @param {Object} text - text to show
+   * @returns {Object}
+   */
+  updating(text) {
+    this.props.onUpdate(text);
```

Comment:
@faboweb Good point but then you would not be notified when a user erased his text.

I got another point. onUpdate is neither in the propTypes required nor you're checking if this is a function.
Everyone using the component without setting onUpdate will get an exception.

Pls reference onUpdate in the propTypes with React.PropTypes.func, define a default value () => {} and check if this is a function before calling. 

```js
static propTypes = {
  onUpdate: React.PropTypes.func
};

static defaultProps = {
  onUpdate: () => {}
}

update(text) {
  if (typeof this.props.onUpdate === 'function') {
    this.props.onUpdate(text);
  }
}
```